workflow-core/src/main/java/com/squareup/workflow/Reactor.kt

```diff
@@ -179,19 +182,57 @@ interface Reactor<S : Any, E : Any, out O : Any> : WorkflowPool.Launcher<S, E, O
 }
 
 /**
- * Use this to implement [WorkflowPool.Launcher.launch].
+ * Use this only to create a top-level workflow. If you're implementing [Launcher.launch], use
+ * [doLaunch].
  */
-fun <S : Any, E : Any, O : Any> Reactor<S, E, O>.doLaunch(
+fun <S : Any, E : Any, O : Any> Reactor<S, E, O>.startRootWorkflow(
   initialState: S,
-  workflows: WorkflowPool,
   context: CoroutineContext = EmptyCoroutineContext
-): Workflow<S, E, O> = ReactorWorkflow(this, initialState, workflows, context)
+): Workflow<S, E, O> = doLaunch(initialState, WorkflowPool(), context)
```

Comment:
(This is why I ❌'d instead of ✔️'d)

Reply:
You don't. This helper doesn't know anything about `WorkflowPool`. `WorkflowPool` is only a `Reactor`-level concept, it assists with writing reactors. All the work it does to provide the next state (but only the next one), that doesn't make sense here, it's only needed because reactors see the world one state at a time.

Reply:
Sorry, confused myself w/the reordering in here.

You know what? startRootWorkflow() is garbage, used only in the integration test. We should get rid of it before it causes harm. I'll get that.